packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2449,6 +2436,22 @@ class TextSelectionGestureDetectorBuilder {
 
       _dragStartViewportOffset = renderEditable.offset.pixels;
       _dragStartScrollOffset = _scrollPosition;
+
+      if (editableText.textEditingValue.selection.isCollapsed && defaultTargetPlatform == TargetPlatform.iOS) {
```

Comment:
nit: I think it probably makes more sense to keep this in the `switch` statement? If the render editable isn't focused, instead of selecting the word, we want to select the position and start the floating cursor right?